mantidproject/mantid
# Muon analysis setting active groups on file load and disabling deselecting all groups/pairs

**Description of work.**
Adds in default selection of groups and pairs when loading a grouping file from xml. Also checks whether a group or pair is the last selected when deselected and disallows the action if so.



**To test:**
 * Load some data into the Muon Analysis GUI.
 * In the grouping tab make a selection of groups and pairs and create a grouping file by clicking save
 * Close the GUI then re-open and re-load the data.
 * Load the new grouping table and check that the rules for default selection described in the issue are followed.
 * Try to unselect all groups and pairs and check deselecting the final group/pair is disabled.



Fixes #29108  




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
